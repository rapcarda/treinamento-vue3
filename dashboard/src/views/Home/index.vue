<template>

    <custom-header
      @create-account="handleAccountCreate"
      @login="handleLogin"
    />

    <contact />

    <div class="felx justify-center py-10 bg-brand-gray">
        <p class="font-medium text-center text-gray-800">feedbacker c 2021</p>
    </div>

</template>

<script>
import { onMounted } from 'vue'
import { useRouter } from 'vue-router'
import CustomHeader from './CustomHeader.vue'
import Contact from './Contact.vue'
import useModal from '../../hooks/useModal'

export default {
  components: { CustomHeader, Contact },
  setup () {
    const router = useRouter()
    const modal = useModal()

    onMounted(() => {
      const token = window.localStorage.getItem('token')
      if (token) {
        // No vue 2 para acessar as rotas utilizava o this.$router, mas no vue3 não tem mais o this. Por isso é importado o useRouter e criado uma const de useRouter
        router.push({ name: 'Feedbacks' })
      }
    })

    function handleLogin () {
      modal.open({
        component: 'ModalLogin'
      })
    }

    function handleAccountCreate () {
      modal.open({
        component: 'ModalCreateAccount'
      })
    }

    return {
      handleLogin,
      handleAccountCreate
    }
  }
}
</script>
