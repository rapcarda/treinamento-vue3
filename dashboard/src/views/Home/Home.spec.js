import Home from '.'
import { shallowMount } from '@vue/test-utils'
// Importa as rotas, pois o home usa
import { routes } from '../../router'
// No index do router, utiliza estes imports abaixo
import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('<Home />', () => {
  it('should render home correctly', async () => {
    // Só vai testar, quando o vue router estiver pronto
    router.push('/')
    await router.isReady()

    // Se estivesse usando vuex, passaria a store aqui também, pois é uma dependência global
    // Tem uma função que chama "mount" que seria utilizada no lugar do "ShallowMount".
    // A diferença é que usando o mount, é montado tudo, incluindo subcomponentes da Home no caso
    // Já o ShallowMount monta somente a home, e ignora os subcomponentes
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
