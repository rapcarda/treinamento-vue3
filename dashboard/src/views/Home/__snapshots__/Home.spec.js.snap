// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home correctly 1`] = `
<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>
<div class="felx justify-center py-10 bg-brand-gray">
  <p class="font-medium text-center text-gray-800">feedbacker c 2021</p>
</div>
`;
